include/caffe/util/gpu_memory.hpp

```diff
@@ -77,7 +71,7 @@ struct GPUMemory {
 
     bool try_reserve(size_t size, int device = INVALID_DEVICE) {
       bool status = true;
-      if (size > size_) {
+      if (size > size_ || ptr_ == NULL) {
```

Comment:
Do we expect ptr to be NULL above this line?
In other words, if ptr is NULL, is that a bug or expected behavior?